pkg/client/devops/jclient/jenkins.go

```diff
@@ -1,49 +1,33 @@
 package jclient
 
 import (
-	"io/ioutil"
-
-	client "github.com/jenkins-zh/jenkins-client/pkg/core"
-	"gopkg.in/yaml.v2"
+	"github.com/jenkins-zh/jenkins-client/pkg/core"
+	"kubesphere.io/devops/pkg/client/devops"
+	"kubesphere.io/devops/pkg/client/devops/jenkins"
 )
 
 type JenkinsClient struct {
-	
-}
-type Config struct {
-	Jenkins JenkinsConfig `yaml:"devops,omitempty"`
-}
-type JenkinsConfig struct {
-	URL      string `yaml:"host"`
-	UserName string `yaml:"username"`
-	Password string `yaml:"password"`
-	Token    string `yaml:"token,omitempty"`
+	Core    core.JenkinsCore
+	jenkins devops.Interface // For refactor purpose only
 }
 
-func GetJenkinsCore() (core client.JenkinsCore, e error) {
-	// read configuration files
-	jenkinsConfigPath := "/etc/kubesphere/kubesphere.yaml"
-	yamlFile, e := ioutil.ReadFile(jenkinsConfigPath)
-	if e != nil {
-		return
-	}
-	// yaml parse
-	var config Config
-	e = yaml.Unmarshal(yamlFile, &config)
-	if e != nil {
-		return
-	}
-	// capsulate JenkinsCore
-	core = client.JenkinsCore{
-		URL:      config.Jenkins.URL,
-		UserName: config.Jenkins.UserName,
-		Token:    config.Jenkins.Password,
+func NewJenkinsClient(options *jenkins.Options) (jenkinsClient *JenkinsClient, err error) {
```

Comment:
Before this line, I suggest to add code as below to ensure `JenkinsClient` implement `devops.Interface` fully:

```go
var _ devops.Interface = &JenkinsClient{}
```